lib/main_page.dart

```diff
@@ -1,88 +1,189 @@
+//   const GenreCard({
+//     super.key,
+//     required this.genre,
+//     required this.onTap,
+//     required this.isSelected,
+//   });
+
+//   final Genre genre;
+//   final VoidCallback onTap;
+//   final bool isSelected;
+
+//   @override
+//   Widget build(context) => GestureDetector(
+//         onTap: onTap,
+//         child: Container(
+//           height: 40,
+//           child: Padding(
+//             padding: const EdgeInsets.all(10),
+//             child: Row(children: [
+//               Text(
+//                 genre.name,
+//                 style: const TextStyle(
+//                   color: Colors.white,
+//                   fontSize: 20,
+//                 ),
+//               ),
+//               Visibility(
+//                 visible: isSelected,
+//                 child: const Icon(
+//                   IconData(
+//                     0xe1f6,
+//                     fontFamily: 'MaterialIcons',
+//                   ),
+//                   color: Colors.white,
+//                   size: 20,
+//                 ),
+//               ),
+//             ]),
+//           ),
+//         ),
+//       );
+// }
+
+// top 10 on IMDb
+class MainPage extends StatefulWidget {
+  const MainPage({super.key});
+
+  @override
+  State<MainPage> createState() => _MainPageState();
+}
+
+class _MainPageState extends State<MainPage> {
+  Set<Movie> _filteredMovies = {};
+  List<Genre> genres =
+      List.generate(Genre.values.length, (index) => Genre.values[index]);
+  final List<bool> _selectedGenres = List.filled(Genre.values.length, false);
+
+  @override
+  void initState() {
+    super.initState();
+    _filteredMovies = Set<Movie>.from(movies);
+  }
+
+  @override
+  Widget build(context) => ScreenWithAppBar(
+        title: 'Top 10 on IMDb',
+        pageContent: Column(
+          children: [
+            Container(
+              height: 60,
+              color: const Color.fromARGB(168, 37, 37, 37),
+              child: Row(
+                children: [
+                  Expanded(
+                    child: ListView.builder(
+                      scrollDirection: Axis.horizontal,
+                      itemCount: genres.length,
+                      itemBuilder: (context, index) => GenreBadge(
+                        genre: genres[index],
+                        isSelected: _selectedGenres[index],
+                        onTap: () {
+                          // for (int i = 0; i < _selectedGenres.length; i++) {
+                          //   _selectedGenres[i] = false;
+                          // }
+                          // _selectedGenres[index] = !_selectedGenres[index];
+                          // _filteredMovies = movies
+                          //     .where((movie) => movie.genre == genres[index])
+                          //     .toList();
+                          // setState(() {});
+                          _selectedGenres[index] = !_selectedGenres[index];
```

Comment:
No need to call setState multiple times, I would rewrite it to something like this:

```
                        onTap: () {
                          if (_selectedGenres.contains(genre)) {
                            _selectedGenres.remove(genre);
                          } else {
                            _selectedGenres.add(genre);
                          }

                          setState(() {
                            if (_selectedGenres.isEmpty) {
                              _filteredMovies = _movies;
                            } else {
                              _filteredMovies =
                                  _movies.where((movie) => _selectedGenres.contains(movie.genre)).toList();
                            }
                          });
                        },
```